src/traces/scattergl/convert.js

```diff
@@ -511,7 +593,11 @@ proto.updateFancy = function(options) {
             this.scatter.options.borderWidths[i] = 0.5 * borderWidths[index];
 
             for(j = 0; j < 4; ++j) {
-                this.scatter.options.colors[4 * i + j] = colors[4 * index + j];
+                var color = colors[4 * index + j];
+                if(sel && !sel[index] && j === 3) {
+                    color *= DESELECTDIM;
+                }
+                this.scatter.options.colors[4 * i + j] = color;
```

Comment:
@dfcreative Something broke selection for _fancy_ traces e.g. with array `marker.size`:

![peek 2017-06-06 17-26](https://user-images.githubusercontent.com/6675409/26852784-68ecfe40-4add-11e7-9e4b-8c0b4fedd68f.gif)

```js
var x = [];
var y = [];
var size = []

for( var i = 0; i < 1e4; i++)
{
  x.push(i);
  y.push(i*Math.random());
  size.push(15 * Math.random())
}

var scatterglMarkers = {
  x: x, y: y, type: "scattergl", mode: "markers",
  marker: {size: size, opacity: 1}
};

console.time('scattergl')
Plotly.newPlot('graph',  [scatterglMarkers], {dragmode: 'lasso'})
.then(() => console.timeEnd('scattergl'))
```